src/metabase/search/appdb/index.clj

```diff
@@ -285,22 +280,24 @@
 (defn reset-index!
   "Ensure we have a blank slate; in case the table schema or stored data format has changed."
   []
-  (drop-table! (pending-table))
+  ;; stop tracking any pending table
+  (when-let [table-name (pending-table)]
+    (when-not *mocking-tables*
+      (search-index-metadata/delete-index! :appdb *index-version-id* table-name))
```

Comment:
why do we delete here instead of marking it retired?

Reply:
The point of retiring tables is so that other instance which are still using that table are not impacted.

This is a "break glass" method, mostly used in dev, for use where we want to force completely new tables. There's no point in complicating it to retired the `active` table while still deleting the `pending` and `retired` ones.